src/map.cpp

```diff
@@ -6038,6 +6038,44 @@ bool map::sees( const tripoint &F, const tripoint &T, const int range, int &bres
     return visible;
 }
 
+int map::obstacle_coverage( const tripoint &loc1, const tripoint &loc2 )
+{
+    if( move_cost( loc2 ) > 2 && !has_flag_ter( TFLAG_FLAT, loc2 ) ) { // Can't hide if you are standing on something.
+        return 0;
+    }
+    tripoint obstaclepos;
+    int dx = loc1.x - loc2.x;
+    int dy = loc1.y - loc2.y;
+    int ax = std::abs( dx ) * 2;
```

Comment:
There's no need for fine grained temps, simply `const int ax = std::abs( loc1.x - loc2.x ) * 2`

Reply:
Thanks, changed.